### DISCUSSION TITLE
Partial redraw

### DISCUSSION THREAD

**Discussion by floers** [rust-windowing/winit#3979] [Q&A]
I want to create an application where it is necessary to rerender only small parts of the window (I want to draw with pen input on an opengl canvas). Obviously this kind of redraw happens when I have e.g. a gtk or slint or whatever ui kit that has damage tracking so there must be a way. I couldn't figure out how to do this.

I think this is what waylands damage_buffer function is used for but I couldn't find a way to do this in a winit window. Is there a way to do this?

**Comment by kchibisov** [comment]
your drawing API should provide it. e.g. softbuffer/EGL/Vulkan/etc.

it's also not about a partial redraw, but rather saying that you did partial redraw, so compositor won't recompose everything. if you want partial redraws, there're different ways for that and APIs, and it's again not a winit issue.
